sklearn/preprocessing/_encoders.py

```diff
@@ -84,6 +104,11 @@ def _fit(self, X, handle_unknown='error'):
             Xi = X_list[i]
             if self.categories == 'auto':
                 cats = _encode(Xi)
+            elif self.categories == 'dtypes':
+                if hasattr(Xi, "cat"):
+                    cats = Xi.cat.categories.values.copy()
+                else:
+                    cats = _encode(Xi)
```

Comment:
The docs imply to me that this setting only handles Categorical dtypes,, not falling back to auto